tests/integration/test_core.py

```diff
@@ -117,6 +116,113 @@ async def test_mysql_relation(ops_test: pytest_operator.plugin.OpsTest, applicat
     )
 
 
+@pytest.mark.asyncio
+async def test_openstack_object_storage_plugin(
+    ops_test: pytest_operator.plugin.OpsTest,
+    application_name,
+    default_admin_password,
+    unit_ip_list,
+    openstack_environment,
+):
+    """
+    arrange: after charm deployed, db relation established and openstack swift server ready.
+    act: update charm configuration for openstack object storage plugin.
+    assert: openstack object storage plugin should be installed after the config update and
+        WordPress openstack swift object storage integration should be set up properly.
+        After openstack swift plugin activated, an image file uploaded to one unit through
+        WordPress media uploader should be accessible from all units.
+    """
+    assert ops_test.model
+    swift_conn = swiftclient.Connection(
```

Comment:
Would it be worth including some comments explaining the key steps being taken here?

Reply:
Added, thanks!

Reply:
That is excellent, it makes a lot of sense now!